common/src/main/java/net/opentsdb/query/QueryNodeConfig.java

```diff
@@ -14,80 +14,82 @@
 // limitations under the License.
 package net.opentsdb.query;
 
-import java.util.List;
-import java.util.Map;
-
 import com.google.common.hash.HashCode;
-
 import net.opentsdb.configuration.Configuration;
 
+import java.util.List;
+import java.util.Map;
+
 /**
  * The configuration interface for a particular query node. Queries will populate
  * the configs when instantiating a DAG.
  * 
  * @since 3.0
  */
-public interface QueryNodeConfig extends Comparable<QueryNodeConfig> {
+public interface QueryNodeConfig<B extends QueryNodeConfig.Builder<B, C>, C extends QueryNodeConfig> extends Comparable<C> {
 
   /**
    * @return The ID of the node in this config.
    */
-  public String getId();
+  String getId();
```

Comment:
Where'd our publics go? Keep em explicit instead of implicit please.

Reply:
Interfaces methods are public by default